terraform/modules/happy-stack-eks/main.tf

```diff
@@ -62,10 +62,21 @@ locals {
   external_services = [for v in var.services : v if v.service_type == "EXTERNAL"]
   internal_services = [for v in var.services : v if v.service_type == "INTERNAL"]
 
+  service_ecrs = { for k, v in module.services : k => v.ecr.repository_url }
+
   task_definitions = { for k, v in var.tasks : k => merge(v, {
     task_name = "${var.stack_name}-${k}"
+    // substitute {service} references in task image with the appropriate ECR repo urls
```

Comment:
This expression comes from https://github.com/hashicorp/terraform/issues/30260, once terraform supports a simpler replacement expression, we will update it.